### PR TITLE
Fix failure when a reference already exists

### PR DESCRIPTION
**Pull request by Javier-varez**
When creating the target branch, if it already exists git2 will return an error with class reference and code exists. We should only return if the error is different than this.

### CODE REVIEW

**Comment by simeg** [comment]
Hey @Javier-varez, thanks for contributing!

~~I'm unable to reproduce this error. Can you describe in more details how the repo should be setup to generate this error?~~

I was able to reproduce it. Maybe it's time to write some integration tests towards this git package..

File: src/git.rs [hunk 1]
```diff
@@ -34,7 +37,13 @@ impl GitManagement for Git {
             .and_then(|oid| repo.find_commit(oid.unwrap()))?;
 
         // Create new branch if it doesn't exist
-        repo.branch(branch_name, &commit, false)?;
+        match repo.branch(branch_name, &commit, false) {
+            // This command can fail due to an existing reference. This error should be ignored.
+            Err(err) if err.class() != ErrorClass::Reference || err.code() != ErrorCode::Exists => {
```
Comment (simeg) [inline src/git.rs]: `||` -> `&&`? We only want to catch this specific error right?

Comment (Javier-varez) [reply src/git.rs]: Exactly, we only want to catch this error, however this does the opposite, let's the error go through. Following De Morgan's laws this code is equivalent to:

```rust
Err(err) if !(err.class() == ErrorClass::Reference && err.code() == ErrorCode::Exists) => {
```

Maybe that would be a bit more readable, I will change it.

**Comment by simeg** [comment]
Awesome, thank you @Javier-varez! Will push a new version soon.

**Pull request merged**